fragments/labels/docker.sh

```diff
@@ -1,13 +1,18 @@
 docker)
     name="Docker"
     type="dmg"
-    if [[ $(arch) == arm64 ]]; then
-     downloadURL="https://desktop.docker.com/mac/stable/arm64/Docker.dmg"
-     appNewVersion=$( curl -fs "https://desktop.docker.com/mac/main/arm64/appcast.xml" | xpath '(//rss/channel/item/enclosure/@sparkle:shortVersionString)[last()]' 2>/dev/null | cut -d '"' -f2 )
-    elif [[ $(arch) == i386 ]]; then
-     downloadURL="https://desktop.docker.com/mac/stable/amd64/Docker.dmg"
-     appNewVersion=$( curl -fs "https://desktop.docker.com/mac/main/amd64/appcast.xml" | xpath '(//rss/channel/item/enclosure/@sparkle:shortVersionString)[last()]' 2>/dev/null | cut -d '"' -f2 )
+    if [[ "$(arch)" == arm64 ]]; then
+        downloadURL="https://desktop.docker.com/mac/stable/arm64/Docker.dmg"
+        appNewVersion=$( curl -fs "https://desktop.docker.com/mac/main/arm64/appcast.xml" | xpath '(//rss/channel/item/enclosure/@sparkle:shortVersionString)[last()]' 2>/dev/null | cut -d '"' -f2 )
+    elif [[ "$(arch)" == i386 ]]; then
+        downloadURL="https://desktop.docker.com/mac/stable/amd64/Docker.dmg"
+        appNewVersion=$( curl -fs "https://desktop.docker.com/mac/main/amd64/appcast.xml" | xpath '(//rss/channel/item/enclosure/@sparkle:shortVersionString)[last()]' 2>/dev/null | cut -d '"' -f2 )
+    fi
+    CLIInstaller="Docker.app/Contents/MacOS/install"
+    CLIArguments=(--accept-license)
+    if [[ "${currentUser}" != "loginwindow" ]];then
+        CLIArguments+=(--user "${currentUser}")
     fi
     expectedTeamID="9BNSXJN65R"
-    blockingProcesses=( "Docker Desktop" "Docker" )
```

Comment:
```sh
$ ps aux | grep -i "docker"
kenchan0130          68931   0.1  0.4 411138208 150352   ??  S     3:12PM   0:00.97 /Applications/Docker.app/Contents/MacOS/com.docker.backend
kenchan0130          68964   0.1  0.2 442390288  64560   ??  S     3:13PM   0:00.43 /Applications/Docker.app/Contents/MacOS/Docker Desktop.app/Contents/Frameworks/Docker Desktop Helper (GPU).app/Contents/MacOS/Docker Desktop Helper (GPU) --type=gpu-process --user-data-dir=/Users/kenchan0130/Library/Application Support/Docker Desktop --enable-features=kWebSQLAccess --disable-features=SpareRendererForSitePerProcess --variations-seed-version --seatbelt-client=58
kenchan0130          60434   0.0  0.1 1585437744  31328   ??  S    月12PM   0:23.05 /Users/kenchan0130/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/kenchan0130/.vscode/extensions/ms-azuretools.vscode-docker-1.29.1/dist/compose-language-service/lib/server.js --node-ipc --node-ipc --clientProcessId=52444
kenchan0130          60433   0.0  0.1 1585437744  31680   ??  S    月12PM   0:21.34 /Users/kenchan0130/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/kenchan0130/.vscode/extensions/ms-azuretools.vscode-docker-1.29.1/dist/dockerfile-language-server-nodejs/lib/server.js --node-ipc --node-ipc --clientProcessId=52444
kenchan0130          69061   0.0  0.0 407962448     48 s004  S+    3:13PM   0:00.00 grep --color=always -i docker
kenchan0130          68968   0.0  0.1 442366048  38352   ??  S     3:13PM   0:00.05 /Applications/Docker.app/Contents/MacOS/Docker Desktop.app/Contents/Frameworks/Docker Desktop Helper.app/Contents/MacOS/Docker Desktop Helper --type=utility --utility-sub-type=network.mojom.NetworkService --lang=ja --service-sandbox-type=network --user-data-dir=/Users/kenchan0130/Library/Application Support/Docker Desktop --standard-schemes=app --secure-schemes=app --fetch-schemes=scout-graphql,docker-hub,docker-extensions-be,project-api --shared-files --field-trial-handle=1718379636,r,17224225255580469348,7527058344799599953,262144 --enable-features=kWebSQLAccess --disable-features=SpareRendererForSitePerProcess --variations-seed-version --seatbelt-client=58
kenchan0130          68948   0.0  0.6 1586311488 195760   ??  S     3:13PM   0:02.44 /Applications/Docker.app/Contents/MacOS/Docker Desktop.app/Contents/MacOS/Docker Desktop --reason=open-tray --analytics-enabled=true --name=dashboard
kenchan0130          68942   0.0  0.1 410860560  29984   ??  S     3:13PM   0:00.06 /Applications/Docker.app/Contents/MacOS/com.docker.virtualization --kernel /Applications/Docker.app/Contents/Resources/linuxkit/kernel --cmdline init=/init loglevel=1 root=/dev/vdb rootfstype=erofs ro vsyscall=emulate panic=0 linuxkit.unified_cgroup_hierarchy=0 console=hvc0   virtio_net.disable_csum=1 eth0.IPNet=192.168.65.3/24 eth0.router=192.168.65.1 eth0.mtu=1500 eth1.dhcp vpnkit.connect=connect://2/1999 --boot /Applications/Docker.app/Contents/Resources/linuxkit/boot.raw --services /Applications/Docker.app/Contents/Resources/linuxkit/services.raw --disk /Users/kenchan0130/Library/Containers/com.docker.docker/Data/vms/0/data/Docker.raw --networkType gvisor --macaddr-filename-prefix vms/0/macaddr --cpus 10 --memoryMiB 8092 --console-log log/vm/console.log --console /Users/kenchan0130/Library/Containers/com.docker.docker/Data/vms/0/console.sock --api /Users/kenchan0130/Library/Containers/com.docker.docker/Data/virtualization.sock --watchdog --kernel-for-udp --virtiofs /Users --virtiofs /Volumes --virtiofs /private --virtiofs /tmp --virtiofs /var/folders
kenchan0130          68941   0.0  0.1 409831792  38624   ??  S     3:13PM   0:00.07 com.docker.build
kenchan0130          68940   0.0  0.1 409796096  37024   ??  S     3:13PM   0:00.05 docker serve --address unix:///Users/kenchan0130/.docker/run/docker-cli-api.sock
kenchan0130          68939   0.0  0.0 409759616  14368   ??  S     3:13PM   0:00.01 com.docker.dev-envs -watchdog
root             68834   0.0  0.0 409777520   7616   ??  Ss    3:12PM   0:00.02 /Library/PrivilegedHelperTools/com.docker.vmnetd
```

```sh
$ pgrep -xq "Docker" && echo "found" || echo "not found"
not found
```

```sh
$ pkill "Docker Desktop"                                                                                                           
$ ps aux | grep -i "docker"
kenchan0130          60434   0.0  0.1 1585437744  31568   ??  S    12PM   0:23.38 /Users/kenchan0130/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/kenchan0130/.vscode/extensions/ms-azuretools.vscode-docker-1.29.1/dist/compose-language-service/lib/server.js --node-ipc --node-ipc --clientProcessId=52444
kenchan0130          60433   0.0  0.1 1585437744  31920   ??  S    12PM   0:21.66 /Users/kenchan0130/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/kenchan0130/.vscode/extensions/ms-azuretools.vscode-docker-1.29.1/dist/dockerfile-language-server-nodejs/lib/server.js --node-ipc --node-ipc --clientProcessId=52444
kenchan0130          77338   0.0  0.0 408636560   1776 s006  S+    3:41PM   0:00.00 grep --color=always -i docker
root             75693   0.0  0.0 409777520   7616   ??  Ss    3:35PM   0:00.02 /Library/PrivilegedHelperTools/com.docker.vmnetd
```

The above results show that there is no process called `Docker` and only `Docker Desktop` needs to be killed.

Since `com.docker.vmnetd` is started via LaunchDaemon, it is ignored because killing it will cause an infinite loop.